src/server/project.ts

```diff
@@ -1728,15 +1729,24 @@ namespace ts.server {
                 return {
                     fileExists: this.program.fileExists,
                     directoryExists: this.program.directoryExists,
-                    realpath: this.program.realpath || this.projectService.host.realpath?.bind(this.projectService.host),
+                    realpath: this.program.realpath || this.projectService.fshost.realpath?.bind(this.projectService.fshost),
                     getCurrentDirectory: this.getCurrentDirectory.bind(this),
-                    readFile: this.projectService.host.readFile.bind(this.projectService.host),
-                    getDirectories: this.projectService.host.getDirectories.bind(this.projectService.host),
+                    readFile: this.projectService.fshost.readFile.bind(this.projectService.fshost),
+                    getDirectories: this.projectService.fshost.getDirectories.bind(this.projectService.fshost),
                     trace: this.projectService.host.trace?.bind(this.projectService.host),
                     useCaseSensitiveFileNames: this.program.useCaseSensitiveFileNames(),
                 };
             }
-            return this.projectService.host;
+            return {

```

Comment:
This should be this.projectService.fsHost

getCurrentDirectory should be on fsHost and not host as its related to FS operations and used to detect which files to include etc.